### PR TITLE
`Exam mode`: Adjust checklist overview for teaching assistants

### PR DESCRIPTION
**Pull request by willergeorg**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I implemented the changes with a good performance and prevented too many database calls.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

We encountered a problem with the checklist overview when a teaching assistant enters the exam management overview. Some data displayed there is fetched from an endpoint that was only accessible for at least instructors.
Omitting the whole checklist would be inordinate as the tutors would benefit from some of the information. On the other hand, the checklist includes some information the tutors should not necessarily have access to. Therefore I decided to adjust the checklist overview for tutors. The overview now displays the following information to the tutor:

- Exam conduction: Start and end date
- Exam correction: 
     - Correction round progress
     - Publish result date
     - Review state
     - Amount of resolved complaints

### Description

I adapted the corresponding endpoint that is returning the statistical data we need especially for the exam conduction part of the overview. It is now accessible for teaching assistants as well, but only includes the adjusted data described above to the response if a tutor sends the request. For instructors, everything should work as before. 
### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Tutor
- 1 Editor
- 1 Exam (feel free to try different states of the exam)

1. Log in to Artemis as Tutor
2. Navigate to Course Administration
3. Navigate to exam management. Make sure the checklist overview loads completely and no errors are shown in the console.
4. Check that the displayed information is correct by checking the checklist overview as instructor.
5. Log in as Editor and check the overview again. The view should be the same as for tutors.
#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

The checklist overview currently looks like this for tutors:
image (6)

How it looks with this PR:
image

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java [hunk 1]
```diff
@@ -282,14 +282,18 @@ private void checkExamCourseIdElseThrow(Long courseId, Exam exam) {
...
+    @PreAuthorize("hasRole('TA')")
     public ResponseEntity<ExamChecklistDTO> getExamStatistics(@PathVariable Long courseId, @PathVariable Long examId) {
         log.debug("REST request to get exam statistics: {}", examId);
 
-        examAccessService.checkCourseAndExamAccessForInstructorElseThrow(courseId, examId);
+        examAccessService.checkCourseAndExamAccessForTeachingAssistantElseThrow(courseId, examId);
+
+        var user = userRepository.getUserWithGroupsAndAuthorities();
+        var course = courseRepository.findByIdElseThrow(courseId);
+        var isInstructorInCourse = authCheckService.isInstructorInCourse(course, user);
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java:293]:
```suggestion
        var course = courseRepository.findByIdElseThrow(courseId);
        var isInstructorInCourse = authCheckService.isAtLeastInstructorInCourse(course, null);
```

**Review by cremertim — DISMISSED** [review]
code changes lgtm.

**Review by valeriebucher — DISMISSED** [review]
Tested on ts1 and worked as expected for every user type!

**Review by Strohgelaender — APPROVED** [review]
code looks good.

**Review by Strohgelaender — APPROVED** [review]
Code looks good (thanks for implementing my requests)
Tested on TS1, the reduced checklist is shown correctly for Tutors. The information is correct and no more errors are present in the console.

**Review by valeriebucher — APPROVED** [review]
Changes look good. The view is shown as expected 👍

**Review by cremertim — APPROVED** [review]
lgtm. reapprove

File: src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java [hunk 2]
```diff
@@ -484,9 +484,10 @@ public void validateForStudentExamGeneration(Exam exam) throws BadRequestAlertEx
      * Gets all statistics for the instructor checklist regarding an exam
      *
      * @param exam the exam for which to get statistics for
+     * @param isInstructor flag indicating if the requesting user is instructor
      * @return a examStatisticsDTO filled with all statistics regarding the exam
      */
-    public ExamChecklistDTO getStatsForChecklist(Exam exam) {
+    public ExamChecklistDTO getStatsForChecklist(Exam exam, boolean isInstructor) {
```
Comment (canberkanar) [inline src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java:490]: Static code analysis warning that seems to be out of this PR's scope.

**Comment by canberkanar** [comment]
Tested locally according to the instructions. Did not see any errors on the console. Compared instructor's view of checklist with tutor and editor, all is fine. Nicely done 👍🏻

**Review by canberkanar — APPROVED** [review]
Tested locally according to the instructions. Did not see any errors on the console. Compared instructor's view of checklist with tutor and editor, all is fine. Nicely done 👍🏻

**Pull request merged**